File: examples/swift/constructs/let_keyword.swift
// Ported from examples/lumen/constructs/let_keyword.lm by scripts/port_examples.py; edit the Lumen original, not this file.
print("Test: let and let mut Keywords")
var x = 10
print("let x = 10")
print(x)
var y = 5
print("let mut y = 5")
print(y)
y = 20
print("After y = 20:")
print(y)
x = 100
print("After let x = 100 (shadowing):")
print(x)
let result = x + y
print("let result = x + y")
print(result)
func test_let() -> Int {
    let a = 42
    var b = 10
    b = 50
    return a + b
}

print("test_let():")
print(test_let())

File: examples/pascal/constructs/let_keyword.pas
// Ported from examples/lumen/constructs/let_keyword.lm by scripts/port_examples.py; edit the Lumen original, not this file.
var x: integer;
var y: integer;
var result: integer;

function test_let(): integer;
var a: integer;
var b: integer;
begin
    a := 42;
    b := 10;
    b := 50;
    test_let := a + b;
end;

begin
    writeln('Test: let and let mut Keywords');
    x := 10;
    writeln('let x = 10');
    writeln(x);
    y := 5;
    writeln('let mut y = 5');
    writeln(y);
    y := 20;
    writeln('After y = 20:');
    writeln(y);
    x := 100;
    writeln('After let x = 100 (shadowing):');
    writeln(x);
    result := x + y;
    writeln('let result = x + y');
    writeln(result);
    writeln('test_let():');
    writeln(test_let());
end.
